network/src/protocols/wire/proxy_protocol.rs

```diff
@@ -0,0 +1,369 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! # An implementation of ProxyProtocol for HAProxy
+//! https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
+//!
+//! ## Limitations
+//! - Only supports ProxyProtocol V2
+//! - Rejects ProxyProtocol V1
+//! - Only supports IPv4 & IPv6 TCP mode
+//! - All other valid connections, will just drop the address information and use the proxy's one
+//!
+//! ## Interpetations not in the spec
+//! - An address space that doesn't match the size expected is rejected e.g. too big for IPv4
+//! - Address space that's larger than the current supported requests is rejected
+
+use crate::transport::TSocket;
+use byteorder::{BigEndian, ReadBytesExt};
```

Comment:
i don't think we need byteorder any more, pretty sure this is included in std now

for example [`std::u16::from_be_bytes`](https://doc.rust-lang.org/std/primitive.u16.html#method.from_be_bytes)

Reply:
I actually did have this, and it didn't compile or something, I don't remember why I switched back to byteorder

Reply:
yeah gotta make sure you pass in a `[u8; X]` to `from_be_bytes` instead of a `&[u8]` slice. note, if you pull in `use bytes::Buf;` you can also read everything into a big buffer and then pull out the pieces like

```rust
let mut buf = [0_u8; 32];
socket.read_exact(&mut buf).await?;

let mut cursor = &mut buf[..];

if cursor.bytes()[..HEADER_SIZE] == &PPV2_SIGNATURE {
    cursor.advance(HEADER_SIZE);
} else {
    Err(..)
}
let version_and_command = cursor.get_u8();
let family_and_protocol = cursor.get_u16();
```

Reply:
I've updated it to not use the `bytes::buf`, but it uses a `try_into` I don't like.

Tbh, this is a lot of overhead and boiler plate just to convert a couple bytes into a number....